components/post_view/post_recent_reactions/post_recent_reactions.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {Dispatch} from 'redux';
+
+import Permissions from 'mattermost-redux/constants/permissions';
+import {Emoji} from 'mattermost-redux/types/emojis';
+import {Locations} from 'utils/constants';
+import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';
+import OverlayTrigger from 'components/overlay_trigger';
+
+import EmojiItem from './recent_reactions_emoji_item';
+
+type LocationTypes = 'CENTER' | 'RHS_ROOT' | 'RHS_COMMENT';
+
+type Props = {
+    channelId?: string;
+    postId: string;
+    teamId: string;
+    getDotMenuRef: () => HTMLDivElement;
+    location: LocationTypes;
+    locale: string;
+    emojis: Emoji[];
+    defaultEmojis: Emoji[];
+    actions: {
+        addReaction: (postId: string, emojiName: string) => (dispatch: Dispatch) => void;
+    };
+}
+
+type State = {
+    location: LocationTypes;
+}
+
+export default class PostRecentReactions extends React.PureComponent<Props, State> {
+    public static defaultProps: Partial<Props> = {
+        location: Locations.CENTER as 'CENTER',
+    };
+
+    handleAddEmoji = (emoji: Emoji): void => {
+        const emojiName = 'short_name' in emoji ? emoji.short_name : emoji.name;
+        this.props.actions.addReaction(this.props.postId, emojiName);
+    };
+
+    complementEmojis(): void {
+        const additional = this.props.defaultEmojis.filter((e) => {
+            let ignore = false;
+            for (const emoji of this.props.emojis) {
+                if (e.name === emoji.name) {
+                    ignore = true;
+                    break;
+                }
+            }
+            return !ignore;
+        });
+        const l = this.props.emojis.length;
+        for (let i = 0; i < 3 - l; i++) {
+            this.props.emojis.push(additional[i]);
+        }
+    }
```

Comment:
we shouldn't mutate props directly, see [this](https://reactjs.org/blog/2015/02/24/streamlining-react-elements.html) for more info.
If it's possible to compliment in the selector that would be best so we avoid duplication as well